### PR TITLE
Update docstrings/documentations of all the datasets

### PR DESCRIPTION
**Pull request by mthrok**
Currently the documentation page does not list constructor arguments or overwrote `__getitem__` methods. This PR fixes it.

(Note: the annotation is broken form the beginning and I cannot fix it.)

Example: 
Screen Shot 2020-10-01 at 18 10 45

### CODE REVIEW

File: torchaudio/datasets/cmuarctic.py [hunk 1]
```diff
@@ -90,7 +96,7 @@ def __init__(self,
                  root: str,
                  url: str = URL,
                  folder_in_archive: str = FOLDER_IN_ARCHIVE,
-                 download: bool = False) -> None:
```
Comment (mruberry) [inline torchaudio/datasets/cmuarctic.py:93]: Interesting: why remove the typing annotation?

File: torchaudio/datasets/cmuarctic.py [hunk 2]
```diff
@@ -76,9 +76,15 @@ def load_cmuarctic_item(line: str,
...
-    waveform, sample_rate, utterance, utterance_id
+    """Create a Dataset for CMU_ARCTIC.
+
+    Args:
+        root (str): Path to the directory where the dataset is found or downloaded.
+        url (str, optional):
+            Type of the dataset to dowload. This is **NOT** the actual URL. (default: ``"aew"``)
+        folder_in_archive (str, optional):
+            The top-level directory of the dataset. (default: ``"ARCTIC"``)
+        download (bool, optional): Download dataset if it is not found at root path. (default: ``False``).
```
Comment (mruberry) [inline torchaudio/datasets/cmuarctic.py:87]: nit: "Whether to download the dataset if ..."

File: torchaudio/datasets/cmuarctic.py (hunk 1 above)
Comment (mthrok) [reply torchaudio/datasets/cmuarctic.py:93]: This adds `-> None` to documentation, and I do not think it adds any value.

File: torchaudio/datasets/cmuarctic.py [hunk 3]
```diff
@@ -76,9 +76,15 @@ def load_cmuarctic_item(line: str,
 
 
 class CMUARCTIC(Dataset):
-    """
-    Create a Dataset for CMU_arctic. Each item is a tuple of the form:
```
Comment (mruberry) [inline torchaudio/datasets/cmuarctic.py:80]: Do you really want to remove the information about the structure of these datasets?

Comment (mthrok) [reply torchaudio/datasets/cmuarctic.py:80]: It's moved to `__getitem__` method.

File: torchaudio/datasets/commonvoice.py [hunk 4]
```diff
@@ -100,11 +100,17 @@ def load_commonvoice_item(line: List[str],
...
-    (waveform, sample_rate, dictionary)
-    where dictionary is a dictionary built from the tsv file with the following keys:
-    client_id, path, sentence, up_votes, down_votes, age, gender, accent.
+    """Create a Dataset for CommonVoice.
+
+    Args:
+        root (str): Path to the directory where the dataset is found or downloaded.
+        tsv (str, optional): The name of the tsv file used to construct the metadata.
+            (default: ``"train.tsv"``)
+        url (str, optional): The name of the language to select the URL when downloading the dataset.
```
Comment (mruberry) [inline torchaudio/datasets/commonvoice.py:109]: Is there a better way to describe this? Maybe "the language of the dataset  to download?"

Comment (mthrok) [reply torchaudio/datasets/commonvoice.py:109]: Ya, that sounds better.

File: torchaudio/datasets/gtzan.py [hunk 5]
```diff
@@ -998,12 +998,20 @@ def load_gtzan_item(fileid: str, path: str, ext_audio: str) -> Tuple[Tensor, str
...
 
-    Please see http://marsyas.info/downloads/datasets.html
-    if you are planning to use this dataset to publish results.
+    Args:
+        root (str): Path to the directory where the dataset is found or downloaded.
+        url (str, optional): The URL to download the dataset from.
+            (default: ``"http://opihi.cs.uvic.ca/sound/genres.tar.gz"``)
+        folder_in_archive (str, optional): The top-level directory of the dataset.
+        download (bool, optional): Download dataset if it is not found at root path. (default: ``False``).
+        subset (str, optional): Subset of the dataset. Either ``"training"``, ``"validation"`` or
```
Comment (mruberry) [inline torchaudio/datasets/gtzan.py:1013]: Maybe: "Which subset of the dataset to use"? 

What happens when subset is None? I assume the full dataset is used. Maybe something like, "Which subset of the dataset to use. Either None, training, validation, or testing. If None the entire dataset is used. (default: None)." ?

File: torchaudio/datasets/cmuarctic.py [hunk 6]
```diff
@@ -76,9 +76,15 @@ def load_cmuarctic_item(line: str,
...
-    waveform, sample_rate, utterance, utterance_id
+    """Create a Dataset for CMU_ARCTIC.
+
+    Args:
+        root (str): Path to the directory where the dataset is found or downloaded.
+        url (str, optional):
+            Type of the dataset to dowload. This is **NOT** the actual URL. (default: ``"aew"``)
```
Comment (mruberry) [inline torchaudio/datasets/cmuarctic.py:84]: How does a reader know which values they can use and what they do?

File: torchaudio/datasets/tedlium.py [hunk 7]
```diff
@@ -140,7 +116,7 @@ def _load_tedlium_item(self, fileid: str, line: int, path: str) -> Tuple[Tensor,
             path (str): Dataset root path
 
         Returns:
-            Tedlium_item: A namedTuple containing [waveform, sample_rate, transcript, talk_id, speaker_id, identifier]
+            tuple: ``(waveform, sample_rate, transcript, talk_id, speaker_id, identifier)``
```
Comment (mruberry) [inline torchaudio/datasets/tedlium.py:119]: Nice fix here.

File: torchaudio/datasets/vctk.py [hunk 8]
```diff
@@ -252,6 +272,14 @@ def _load_sample(self, speaker_id: str, utterance_id: str, mic_id: str) -> Sampl
         return Sample(waveform, sample_rate, utterance, speaker_id, utterance_id)
 
     def __getitem__(self, n: int) -> Sample:
+        """Load the n-th sample from the dataset.
```
Comment (mruberry) [inline torchaudio/datasets/vctk.py:275]: Interesting that the tuple description is kept on the class docstring for this dataset.

File: torchaudio/datasets/cmuarctic.py (hunk 6 above)
Comment (mthrok) [reply torchaudio/datasets/cmuarctic.py:84]: Yeah, so that's the problem. I see the list of valid values like the followings and I wonder, what is the best way to add all of them???????

```
        if url in [
            "aew",
            "ahw",
            "aup",
            "awb",
            "axb",
            "bdl",
            "clb",
            "eey",
            "fem",
            "gka",
            "jmk",
            "ksp",
            "ljm",
	    "lnh",
            "rms",
            "rxr",
            "slp",
            "slt"
	]:
```

File: torchaudio/datasets/cmuarctic.py (hunk 1 above)
Comment (mruberry) [reply torchaudio/datasets/cmuarctic.py:93]: We actually had an issue recently in PyTorch to add these "-> None" typing annotations to our inits. Check out:

https://github.com/pytorch/pytorch/issues/45497

You might save yourself a similar issue by leaving them in.

File: torchaudio/datasets/vctk.py (hunk 8 above)
Comment (mthrok) [reply torchaudio/datasets/vctk.py:275]: Oh this is because we intend to change NamedTuple to a normal tuple for consistency reason before the release. (so very soon) So this way I do not have to change the docstrings twice.

File: torchaudio/datasets/cmuarctic.py (hunk 1 above)
Comment (mthrok) [reply torchaudio/datasets/cmuarctic.py:93]: oh good to know. Okay I will revert it.

File: torchaudio/datasets/gtzan.py (hunk 5 above)
Comment (mthrok) [reply torchaudio/datasets/gtzan.py:1013]: Yes, your assumption is correct. Updated the description.

File: torchaudio/datasets/cmuarctic.py (hunk 6 above)
Comment (mthrok) [reply torchaudio/datasets/cmuarctic.py:84]: okay added them as is.

File: torchaudio/datasets/vctk.py (hunk 8 above)
Comment (mthrok) [reply torchaudio/datasets/vctk.py:277]: oh I miss red your comment. it's an oversight. I will remove it from there.

File: torchaudio/datasets/cmuarctic.py (hunk 3 above)
Comment (mthrok) [reply torchaudio/datasets/cmuarctic.py:80]: I think describing it in `__getitem__` has better locality, which is better for the developer experience. If something is changed on the implementation side, they only need to change the corresponding docstring and there are less chances that there are discrepancy between the docstring description and the actual implementation, compared to the case where the same information are repeated in class description.
Of course, we are having the current docathon to catch such discrepancies, so one can argue that it's okay to have such discrepancies, but it will confuse users on master branch still.

File: torchaudio/datasets/cmuarctic.py [hunk 9]
```diff
@@ -76,9 +76,19 @@ def load_cmuarctic_item(line: str,
...
+            The valid choice is ``"aew"``, ``"ahw"``, ``"aup"``, ``"awb"``, ``"axb"``, ``"bdl"``,
```
Comment (mruberry) [inline torchaudio/datasets/cmuarctic.py:85]: Nit: "The valid choice is" -> "Allowed url values are ..."

File: torchaudio/datasets/speechcommands.py [hunk 10]
```diff
@@ -36,9 +36,16 @@ def load_speechcommands_item(filepath: str, path: str) -> Tuple[Tensor, int, str
...
 class SPEECHCOMMANDS(Dataset):
-    """
-    Create a Dataset for Speech Commands. Each item is a tuple of the form:
-    waveform, sample_rate, label, speaker_id, utterance_number
+    """Create a Dataset for Speech Commands.
+
+    Args:
+        root (str): Path to the directory where the dataset is found or downloaded.
+        url (str, optional): Type of the dataset to dowload. This is **NOT** the actual URL.
+            (default: ``"speech_commands_v0.02"``)
```
Comment (mruberry) [inline torchaudio/datasets/speechcommands.py:44]: This could also use a more complete enumeration of the options (and possibly what each means). That can be a separate PR, too.

**Review by mruberry — APPROVED** [review]
This changes look like a significant improvement.

In the future it may be interesting to enumerate all supported inputs for each param along with a brief description of what they do.

File: docs/source/datasets.rst [hunk 11]
```diff
@@ -29,82 +29,85 @@ CMUARCTIC
 ~~~~~~~~~
 
 .. autoclass:: CMUARCTIC
-  :members: __getitem__
-  :special-members:
+  :members:
+  :special-members: __getitem__
```
Comment (vincentqb) [inline docs/source/datasets.rst:33]: nit: out of curiosity, does this change the output?

File: torchaudio/datasets/libritts.py [hunk 12]
```diff
@@ -65,9 +65,16 @@ def load_libritts_item(
...
 
 class LIBRITTS(Dataset):
-    """
-    Create a Dataset for LibriTTS. Each item is a tuple of the form:
-    waveform, sample_rate, original_text, normalized_text, speaker_id, chapter_id, utterance_id
+    """Create a Dataset for LibriTTS.
+
+    Args:
+        root (str): Path to the directory where the dataset is found or downloaded.
+        url (str, optional): Type of the dataset to dowload. This is **NOT** the actual URL.
```
Comment (vincentqb) [inline torchaudio/datasets/libritts.py:72]: same about code path for url

File: torchaudio/datasets/speechcommands.py [hunk 13]
```diff
@@ -36,9 +36,16 @@ def load_speechcommands_item(filepath: str, path: str) -> Tuple[Tensor, int, str
...
 
 class SPEECHCOMMANDS(Dataset):
-    """
-    Create a Dataset for Speech Commands. Each item is a tuple of the form:
-    waveform, sample_rate, label, speaker_id, utterance_number
+    """Create a Dataset for Speech Commands.
+
+    Args:
+        root (str): Path to the directory where the dataset is found or downloaded.
+        url (str, optional): Type of the dataset to dowload. This is **NOT** the actual URL.
```
Comment (vincentqb) [inline torchaudio/datasets/speechcommands.py:43]: same about url

File: torchaudio/datasets/commonvoice.py [hunk 14]
```diff
@@ -100,11 +100,27 @@ def load_commonvoice_item(line: List[str],
...
-    (waveform, sample_rate, dictionary)
-    where dictionary is a dictionary built from the tsv file with the following keys:
-    client_id, path, sentence, up_votes, down_votes, age, gender, accent.
+    """Create a Dataset for CommonVoice.
+
+    Args:
+        root (str): Path to the directory where the dataset is found or downloaded.
+        tsv (str, optional): The name of the tsv file used to construct the metadata.
+            (default: ``"train.tsv"``)
+        url (str, optional): The language of the dataset to download.
```
Comment (vincentqb) [inline torchaudio/datasets/commonvoice.py:109]: same about url

File: torchaudio/datasets/cmuarctic.py [hunk 15]
```diff
@@ -76,9 +76,19 @@ def load_cmuarctic_item(line: str,
...
+            ``"clb"``, ``"eey"``, ``"fem"``, ``"gka"``, ``"jmk"``, ``"ksp"``, ``"ljm"``, ``"lnh"``,
+            ``"rms"``, ``"rxr"``, ``"slp"`` or ``"slt"``.
```
Comment (vincentqb) [inline torchaudio/datasets/cmuarctic.py:87]: same about url

File: torchaudio/datasets/speechcommands.py (hunk 10 above)
Comment (mthrok) [reply torchaudio/datasets/speechcommands.py:46]: Updated as part of this PR.

File: torchaudio/datasets/librispeech.py [hunk 16]
```diff
@@ -67,9 +67,16 @@ def load_librispeech_item(fileid: str,
...
 
 class LIBRISPEECH(Dataset):
-    """
-    Create a Dataset for LibriSpeech. Each item is a tuple of the form:
-    waveform, sample_rate, utterance, speaker_id, chapter_id, utterance_id
+    """Create a Dataset for LibriSpeech.
+
+    Args:
+        root (str): Path to the directory where the dataset is found or downloaded.
+        url (str, optional): Type of the dataset to dowload. This is **NOT** the actual URL.
```
[the remark this answers is not in the document]
Comment (mthrok) [reply torchaudio/datasets/librispeech.py]: oh you are right. updated.

**Review by vincentqb — APPROVED** [review]
thanks for updating this, lgtm overall :)
